stacker_blueprints/alb.py

```diff
@@ -0,0 +1,283 @@
+        },
+        'VpcId': {
+            'type': str,
+            'description': 'The physical ID of the VPC.'
+        },
+        'TargetGroups': {
+            'type': list,
+            'description': 'List of target groups to attach to the load balancer.'
+        },
+        'Listeners': {
+            'type': list,
+            'description': 'List of listeners for the load balancer.'
+        },
+        'Alarms': {
+            'type': list,
+            'description': 'List of alarms for the load balancer.'
+        },
+    }
+
+    def create_alarm(self, alarm_name, alarm):
+        t = self.template
+
+        alarm.setdefault('Namespace', 'AWS/ApplicationELB')
+        alarm_name = alarm_name.replace('_', '').replace('-', '')
+
+        alarm_resource = t.add_resource(
+            cloudwatch.Alarm.from_dict(alarm_name, alarm)
+        )
+
+        t.add_output(
+            Output(alarm_name, Value=alarm_resource.Ref())
+        )
+
+        t.add_output(
+            Output(alarm_name + "Arn", Value=alarm_resource.GetAtt("Arn"))
+        )
+
+    def create_load_balancer(self):
+        t = self.template
+        v = self.get_variables()
+
+        ingress_rules = v['IngressRules']
+        egress_rules = v['EgressRules']
+        security_groups = v['SecurityGroups']
+        tags = v['Tags']
+        tags.setdefault('Name', v['Name'])
+
+        if ingress_rules or egress_rules:
+            name = '{}-alb'.format(v['Name'])
+            security_group = t.add_resource(
+                ec2.SecurityGroup(
+                    'SECGROUP',
+                    GroupName=name,
+                    GroupDescription='Security group for {}'.format(v['Name']),
+                    SecurityGroupIngress=ingress_rules,
+                    SecurityGroupEgress=egress_rules,
+                    Tags=Tags({
+                        'Name': name,
+                    }),
+                    VpcId=v['VpcId'],
+                )
+            )
+            security_groups.append(security_group.Ref())
+
+        logical_name = 'ALB'
+
+        load_balancer = t.add_resource(
+            alb.LoadBalancer(
+                'ALB',
+                Name=v['Name'],
+                Scheme=v['Scheme'],
+                IpAddressType=v['IpAddressType'],
+                SecurityGroups=security_groups,
+                Subnets=v['SubnetIds'],
+                Tags=Tags(tags),
+                Type=v['Type'],
+            )
+        )
+
+        alarms = v.get("Alarms", [])
+        for alarm in alarms:
+            alarm_name = '{}Alarm{}'.format(logical_name, alarm['MetricName'])
+
+            alarm['Dimensions'] = [
+                {
+                    'Name': 'LoadBalancer',
+                    'Value': load_balancer.GetAtt('LoadBalancerFullName'),
+                },
+            ]
+
+            self.create_alarm(alarm_name, alarm)
+
+        t.add_output(
```

Comment:
A small update to stacker recently gave a helper that might be good to use in here - the blueprint base class in 1.4 has an `add_output` helper:

```
self.add_output("LoadBalancerArn", load_balancer.Ref())
```